internal/commands/cache_compiled_releases_test.go

```diff
@@ -252,6 +252,125 @@ func TestCacheCompiledReleases_Execute_when_one_release_is_cached_another_is_alr
 	}))
 }
 
+// this test covers
+// - when a release is compiled with a stemcell that is not the one in the Kilnfile.lock (aka the compilation target)
+// - the deployment succeeds because the stemcell major lines are the same/compatible
+// - export release returns a broken bosh release because we requested the wrong compilation target and the director didn't have the source code necessarily to re-compile against the requested stemcell
+// - (ideally bosh export-rlease should return an error but in this case it doesn't so we are just checking for a release with the correct stemcell before downloading a bad one)
+func TestCacheCompiledReleases_Execute_when_a_release_is_not_compiled_with_the_correct_stemcell(t *testing.T) {
+	please := Ω.NewWithT(t)
+
+	// setup
+
+	fs := memfs.New()
+
+	please.Expect(fsWriteYAML(fs, "Kilnfile", cargo.Kilnfile{
+		ReleaseSources: []cargo.ReleaseSourceConfig{
+			{
+				ID:           "cached-compiled-releases",
+				Publishable:  true,
+				PathTemplate: "{{.Release}}-{{.Version}}.tgz",
+			},
+			{
+				ID:           "new-releases",
+				Publishable:  false,
+				PathTemplate: "{{.Release}}-{{.Version}}.tgz",
+			},
+		},
+	})).NotTo(Ω.HaveOccurred())
+	please.Expect(fsWriteYAML(fs, "Kilnfile.lock", cargo.KilnfileLock{
+		Releases: []cargo.ComponentLock{
+			{
+				Name:    "banana",
+				Version: "2.0.0",
+
+				RemoteSource: "cached-compiled-releases",
+				RemotePath:   "banana-2.0.0-alpine-5.5.5",
+
+				SHA1: "fake-checksum",
+			},
+		},
+		Stemcell: cargo.Stemcell{
+			OS:      "alpine",
+			Version: "8.0.0",
+		},
+	})).NotTo(Ω.HaveOccurred())
+
+	opsManager := new(fakes.OpsManagerReleaseCacheSource)
+	opsManager.GetStagedProductManifestReturns(`{"name": "cf-some-id", "stemcells": [{"os": "alpine", "version": "8.0.0"}]}`, nil)
+
+	releaseInBlobstore := []byte(`lemon-release-buffer`)
+
+	deployment := new(boshdirFakes.FakeDeployment)
+	bosh := new(boshdirFakes.FakeDirector)
+	bosh.FindDeploymentReturns(deployment, nil)
+	bosh.HasReleaseReturns(false, nil)
```

Comment:
We should put a comment or something that calls out that this is the important failure line